config/passport.js

```diff
@@ -0,0 +1,79 @@
+const GoogleStrategy = require('passport-google-oauth').OAuth2Strategy;
+const LocalStrategy = require('passport-local').Strategy;
+const BearerStrategy = require('passport-http-bearer').Strategy;
+const configAuth = require('./settings');
+
+module.exports = function(passport, models) {
+
+	const localUser = models.user_account;
+
+	passport.serializeUser(function(user, done){
+		done(null, user);
+	});
+
+	passport.deserializeUser(function(user, done){
+		done(null, user);
+	});
+
+	passport.use(new GoogleStrategy({
+			clientID: configAuth.googleAuth.clientID,
+			clientSecret: configAuth.googleAuth.clientSecret,
+			callbackURL: configAuth.googleAuth.callbackURL,
+			access_type: configAuth.googleAuth.access_type
+		},
+	  	function(req, accessToken, refreshToken, profile, done) {
+	  		process.nextTick(function(){
+	  			localUser.findOrCreate({where: {
+	  				email: profile.emails[0].value,
+	  				provider: 'google'
+	  			}, defaults:{
+	  				email: profile.emails[0].value,
+	  				provider: 'google',
+	  				google_token: accessToken,
+	  				google_id: profile.id,
+	  				name: profile.displayName
+	  			}})
+	  			.spread((user, created)=> {
+	  				if(!created && !user){
```

Comment:
I am using a single user table for google and local users, now supposing a user does not sign in with google and registers an account with his/her Gmail ID, then I prevent such users from signing in through google sign in as they are already registered through the local registration.